meroxa/resource_connector.go

```diff
@@ -283,3 +295,47 @@ func resourceConnectorConfig(d *schema.ResourceData) map[string]interface{} {
 
 	return config
 }
+
+func validateConnectorName() schema.SchemaValidateDiagFunc {
+	return func(val interface{}, path cty.Path) diag.Diagnostics {
+		var diags diag.Diagnostics
+		name := val.(string)
+
+		if len(name) > connectorNameMax {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Invalid connector name",
+				Detail:   fmt.Sprintf("connector name should not be longer than %d characters", connectorNameMax),
+			})
```

Comment:
I took inspiration from other `diag.Diagnostics` uses in this repo.

Is it possible to append multiple errors to the same slice here? It would be nice from the perspective of knowing multiple reasons why a name is invalid, rather than having to try over and over.